src/main/java/org/folio/tlib/postgres/impl/PgCqlQueryImpl.java

```diff
@@ -0,0 +1,315 @@
+import org.z3950.zing.cql.CQLSortNode;
+import org.z3950.zing.cql.CQLTermNode;
+
+public class PgCqlQueryImpl implements PgCqlQuery {
+  private static final Logger log = LogManager.getLogger(PgCqlQueryImpl.class);
+
+  final CQLParser parser = new CQLParser(CQLParser.V1POINT2);
+  final Map<String, PgCqlField> fields = new HashMap<>();
+
+  String language = "english";
+  CQLNode cqlNodeRoot;
+
+  @Override
+  public void parse(String query) {
+    if (query == null) {
+      cqlNodeRoot = null;
+    } else {
+      try {
+        log.debug("Parsing {}", query);
+        cqlNodeRoot = parser.parse(query);
+      } catch (CQLParseException | IOException e) {
+        throw new IllegalArgumentException(e.getMessage());
+      }
+    }
+  }
+
+  @Override
+  public String getWhereClause() {
+    return handleWhere(cqlNodeRoot);
+  }
+
+  static String basicOp(CQLTermNode termNode) {
+    String base = termNode.getRelation().getBase();
+    switch (base) {
+      case "==":
+        return "=";
+      case "=":
+      case "<>":
+        return base;
+      default:
+        throw new IllegalArgumentException("Unsupported operator " + base + " for: "
+            + termNode.toCQL());
+    }
+  }
+
+  static String numberOp(CQLTermNode termNode) {
+    String base = termNode.getRelation().getBase();
+    switch (base) {
+      case "==":
+        return "=";
+      case "=":
+      case "<>":
+      case ">":
+      case "<":
+      case "<=":
+      case ">=":
+        return base;
+      default:
+        throw new IllegalArgumentException("Unsupported operator " + base + " for: "
+            + termNode.toCQL());
+    }
+  }
+
+  /**
+   * See if this is a CQL query with a existence check (NULL or NOT NULL).
+   * <p>Empty term makes "IS NULL" for CQL relation =, "IS NOT NULL" for CQL relation <>.
+   * </p>
+   * @param field CQL field.
+   * @param termNode term.
+   * @return SQL op for NULL; null if not a NULL check.
+   */
+  String handleNull(PgCqlField field, CQLTermNode termNode) {
+    if (!termNode.getTerm().isEmpty()) {
+      return null;
+    }
+    String base = termNode.getRelation().getBase();
+    switch (base) {
+      case "=":
+        return field.getColumn() + " IS NULL";
+      case "<>":
+        return field.getColumn() + " IS NOT NULL";
+      default:
+        return null;
+    }
+  }
+
+  String handleTypeUuid(PgCqlField field, CQLTermNode termNode) {
+    String s = handleNull(field, termNode);
+    if (s != null) {
+      return s;
+    }
+    UUID id = UUID.fromString(termNode.getTerm()); // so IllegalArgumentException is thrown
+    String pgTerm = "'" + id + "'";
+    String op = basicOp(termNode);
+    return field.getColumn() + op + pgTerm;
+  }
+
+  /**
+   * Convert CQL term to Postgres term - exact - without C style escapes in result.
+   * <p> Double backslash is converted to backslash. Postgres quotes (') are escaped.
+   * Otherwise things are passed through verbatim.
```

Comment:
This assumes the `unmasked` modifier.
However, CQL defines that the `masked` modifier is the default:
https://www.loc.gov/standards/sru/cql/contextSets/theCqlContextSet.html

It deviates from the CQL spec and from the way RMB implements it. Is this intended?

Please add a README or similar document that explains the masked/unmasked behavior and how it differs from RMB and the CQL spec.

We already have a well tested CQL string to SQL string converter:
https://github.com/folio-org/raml-module-builder/blob/v33.0.2/cql2pgjson/src/main/java/org/folio/cql2pgjson/util/Cql2SqlUtil.java#L86-L128
https://github.com/folio-org/raml-module-builder/blob/v33.0.2/cql2pgjson/src/test/java/org/folio/cql2pgjson/util/Cql2SqlUtilTest.java#L101-L130

Reply:
This is you and me differ. I don't want to repeat code that's already in Postgres. Just enough to ensure no Postgres injection, but of course letting all VALID numbers pass through.